plugins/shorewall/shorewall_log

```diff
@@ -0,0 +1,292 @@
+                 - read all tags from iptables config, instead of the last 24h
+                   of logs
+                 - add support for journald
+                 - use cursors for accuracy
+                 - convert to multigraph to reduce load
+
+=head1 USAGE
+
+Parameters understood:
+
+  config   (required)
+  autoconf (optional - used by munin-config)
+
+=head1 MAGIC MARKERS
+
+#%# family=auto
+#%# capabilities=autoconf
+"""
+
+
+import sys
+import os
+
+
+def autoconf() -> str:
+    if sys.version_info < (3, 5):
+        return 'no (This plugin requires python 3.5 or higher)'
+
+    if os.getenv('MUNIN_CAP_MULTIGRAPH', '0') != '1':
+        return 'no (No multigraph support)'
+
+    import shutil
+
+    if not shutil.which('shorewall'):
+        return 'no (No shorewall executable found)'
+
+    if not shutil.which('iptables-save'):
+        return 'no (No iptables-save executable found, required for tag enumeration)'
+
+    return 'yes'
+
+
+if len(sys.argv) == 2 and sys.argv[1] == "autoconf":
+    print(autoconf())
+    sys.exit()
+
+
+from collections import defaultdict, namedtuple
+from typing import Set, Iterator, TextIO, Dict, Tuple
+from itertools import takewhile
+from subprocess import run, PIPE
+import shlex
+import shutil
+import pickle
+import re
+
+
+logfile = os.getenv('logfile', 'journald' if shutil.which('journalctl') else '/var/log/kern.log')
+journalctl_args = list(shlex.split(os.getenv('journalctlargs',
+                                             'SYSLOG_IDENTIFIER=kernel')))
+taggroups = os.getenv('taggroups')
+taggroups = taggroups.split() if taggroups else []
+tagfilter = os.getenv('tagfilter')
+tagfilter = tagfilter.split() if tagfilter else []
+include_ungrouped = (os.getenv('includeungrouped', 'true').lower() == 'true')
+prefix_format = os.getenv('prefixformat')
+if prefix_format:
+    if sys.version_info < (3, 7):
+        prefix_format = re.escape(prefix_format).replace('\\%s', '(.+)').replace('\\%d', '\\d+')
+    else:  # % is no longer escaped
+        prefix_format = re.escape(prefix_format).replace('%s', '(.+)').replace('%d', '\\d+')
+    prefix_format = re.compile(prefix_format)
+
+
+def get_logtags() -> Tuple[Dict[str, Set[str]], Dict[str, Set[str]]]:
+    rules = (run(['iptables-save'], stdout=PIPE, universal_newlines=True).
+             stdout.splitlines())
+    tags = defaultdict(set)
+    groups = defaultdict(set)
+
+    for line in rules:
+        args = iter(shlex.split(line))
+        for arg in args:
+            if arg == '-j' and next(args) != 'LOG':
+                break
+            if arg == '--log-prefix':
+                prefix = next(args)
+
+                if prefix_format:
+                    tag = prefix_format.match(prefix)[1]
+                else:
+                    tag = prefix.rstrip()
+
+                for ignored in tagfilter:
+                    if ignored in tag:
+                        continue
```

Comment:
I think, this will just leave the inner `for` loop.
How about the following?
```python
if any(ignored in tag for ignored in tagfilter):
    continue
```